libsolidity/formal/SymbolicState.cpp

```diff
@@ -16,24 +16,43 @@
 */
 // SPDX-License-Identifier: GPL-3.0
 
+#include "libsmtutil/SolverInterface.h"
 #include <libsolidity/formal/SymbolicState.h>
 
+#include <libsolidity/formal/SymbolicTypes.h>
 #include <libsolidity/formal/EncodingContext.h>
 
 using namespace std;
 using namespace solidity;
+using namespace solidity::smtutil;
 using namespace solidity::frontend::smt;
 
 SymbolicState::SymbolicState(EncodingContext& _context):
 	m_context(_context)
 {
+	m_stateMembers.emplace("balances", make_shared<smtutil::ArraySort>(smtutil::SortProvider::uintSort, smtutil::SortProvider::uintSort));
+
+	unsigned i = 0;
+	vector<string> members;
+	vector<SortPointer> sorts;
+	for (auto const& [component, sort]: m_stateMembers)
+	{
+		members.emplace_back(component);
+		sorts.emplace_back(sort);
+		m_componentIndices[component] = i++;
```

Comment:
Or maybe even removing `i` and doing something like:
```suggestion
		m_componentIndices[component] = members.size() - 1;
```